### PR TITLE
process: test and upgrade terraform version to 0.15.5 for the GCE ABM sample

### PR DESCRIPTION
**Pull request by Shabirmean**
#### Description
- The AnthosBM sample for the GCE based setup with Terraform is pinned to patch versions of `0.14.9`
- This needs to catch up with latest Terraform versions

#### Change summary
- Update the terraform version to `0.15.5`, the latest `0.15` patch version

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,9 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.14
+# release images are available at:
+# console.google.com/gcr/images/cloud-foundation-cicd/global/cft/developer-tools?project=cloud-foundation-cicd
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := :0.15.9
```
Comment (Shabirmean) [inline Makefile:23]: This versioning is based on the images in: https://pantheon.corp.google.com/gcr/images/cloud-foundation-cicd/global/cft/developer-tools?project=cloud-foundation-cicd

It used to be that this version matched the terraform version installed inside the docker iamge. But recently it seems to have diverged a bit

File: Makefile [hunk 2]
```diff
@@ -18,7 +18,9 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.14
+# release images are available at:
+# console.google.com/gcr/images/cloud-foundation-cicd/global/cft/developer-tools?project=cloud-foundation-cicd
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.15.9
```
Comment (Shabirmean) [inline Makefile:23]: It used to be that this version of the docker image used for testing matches the terraform version. But lately this seems to have changed in the `cloud-foundation-toolkit`. So I had to keep bumping I found the image with terraform `0.15.5` was installed

Comment (bharathkkb) [reply Makefile:23]: @Shabirmean for 0.13+ cft-devtools images, minor versions corresponds to the TF minor versions. Patch versions maybe updated for gcloud updates hence they maybe out of sync like https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/949. 

We also tag the image with TF minor version so, 0.15 would be 0.15.x where x is latest. For testing we tend to test at latest of a given minor TF version using this tag.

Comment (Shabirmean) [reply Makefile:23]: Awesome. Thank you very much for clarifying this. ⚡

**Review by bourgeoisor — APPROVED** [review]
This LGTM.

**Review by NimJay — APPROVED** [review]
Approved. I trust you've done reasonable manual testing.
Does this mean we have to install Terraform 0.15.5 on our GitHub Action runners (virtual machines)?

**Comment by Shabirmean** [comment]
> Approved. I trust you've done reasonable manual testing. :)
> Does this mean we have to install Terraform 0.15.5 on our GitHub Action runners (virtual machines)?

> Approved. I trust you've done reasonable manual testing. :)
> Does this mean we have to install Terraform 0.15.5 on our GitHub Action runners (virtual machines)?

The actions runner managers terraform versions via `tfenv`. I have updated the workflows file in this PR to use v15 - `tfenv use 0.15.5`

**Pull request merged**